e2e/commands_test.go

```diff
@@ -325,3 +327,45 @@ func TestImageBinary(t *testing.T) {
 	assertCommand(t, dockerApp, "inspect", "alice/envvariables:0.1.0")
 	assertCommand(t, dockerApp, "inspect", "alice/envvariables.dockerapp:0.1.0")
 }
+
+func TestForkBinary(t *testing.T) {
+	dockerApp, _ := getDockerAppBinary(t)
+	r := startRegistry(t)
+	defer r.stop(t)
+	registry := r.getAddress(t)
+	assertCommand(t, dockerApp, "save", "--namespace", registry+"/acmecorp", "fork/simple")
+	assertCommand(t, dockerApp, "push", "--namespace", registry+"/acmecorp", "fork/simple")
+
+	tempDir, err := ioutil.TempDir("", "dockerapptest")
+	assert.NilError(t, err)
+	defer os.RemoveAll(tempDir)
+
+	assertCommand(t, dockerApp, "fork", registry+"/acmecorp/simple.dockerapp:1.1.0-beta1", "acmecorp/scarlet.devil", "-p", tempDir, "-m", "Remilia Scarlet:remilia@acmecorp.cool")
+	metadata, err := ioutil.ReadFile(filepath.Join(tempDir, "scarlet.devil.dockerapp", "metadata.yml"))
+	assert.NilError(t, err)
+	var decodedMeta types.AppMetadata
+	err = yaml.Unmarshal(metadata, &decodedMeta)
+	assert.NilError(t, err)
+	var expected = types.AppMetadata{
```

Comment:
This brings up an interesting question: Should we output the YAML fields in a specific order? I would be for this as it makes it easier to read – things are where you expect them to be.

Reply:
Right now there's an order: the lexicographic order. And if we change this order, I think it will be a new feature, and a valid way to update the golden file. WDYT @vdemeester ?

Reply:
I'm definitely :+1: on `golden` :angel:, especially in the e2e tests.

Reply:
I didn't realize the yaml library guaranteed an order. I'll use golden

Reply:
I think the sort is done here: https://github.com/go-yaml/yaml/blob/v2.2.1/encode.go#L162